### PR TITLE
Remove log_entries from object deletion logic, detach documents from them on delete

### PR DESCRIPTION
**Pull request by blms**
## What this PR does

- Overrides admin `get_deleted_objects` so that log entries are removed from the pre-delete check
- Detaches documents from log entries before a document is deleted
- Shows delete confirmation message on "delete selected" in addition to "delete a single document"
- Per #537:
  - Deletes `DocumentPrefetchableProxy` model
  - In a migration, reassigns all `LogEntry` objects from the proxy model to the original `Document` model

### CODE REVIEW

File: geniza/corpus/migrations/0026_delete_documentprefetchableproxy.py [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+    ]
+
+    operations = [
+        # delete DocumentPrefetchableProxy model
+        migrations.DeleteModel(
+            name="DocumentPrefetchableProxy",
+        ),
+        # reassign all log entries associated with DocumentPrefetchableProxy
+        # instances to Documents instead
+        migrations.RunPython(reassign_log_entries, migrations.RunPython.noop),
```
Comment (rlskoeser) [inline geniza/corpus/migrations/0026_delete_documentprefetchableproxy.py:32]: does this order work fine? the prefetchable proxy content type isn't deleted when the model is deleted, right?

File: geniza/corpus/templates/admin/corpus/document/delete_selected_confirmation.html [hunk 2]
```diff
@@ -0,0 +1,6 @@
+{% extends "admin/delete_selected_confirmation.html" %}
```
Comment (rlskoeser) [inline geniza/corpus/templates/admin/corpus/document/delete_selected_confirmation.html:1]: is this one for deleting multiple documents? good catch, thanks for generalizing the message

File: geniza/corpus/tests/test_corpus_models.py [hunk 3]
```diff
@@ -1,4 +1,5 @@
 from datetime import datetime
+from multiprocessing.dummy import freeze_support
```
Comment (rlskoeser) [inline geniza/corpus/tests/test_corpus_models.py:2]: what's this?

**Review by rlskoeser — APPROVED** [review]
Looks good. Just a couple of small questions about migration order and a test, and one import that looks unnecessary.

File: geniza/corpus/tests/test_corpus_models.py (hunk 3 above)
Comment (blms) [reply geniza/corpus/tests/test_corpus_models.py:2]: Oops, looks like an intellisense mistake that I didn't catch, thanks!

File: geniza/corpus/templates/admin/corpus/document/delete_selected_confirmation.html (hunk 2 above)
Comment (blms) [reply geniza/corpus/templates/admin/corpus/document/delete_selected_confirmation.html:1]: Exactly!

File: geniza/corpus/migrations/0026_delete_documentprefetchableproxy.py (hunk 1 above)
Comment (blms) [reply geniza/corpus/migrations/0026_delete_documentprefetchableproxy.py:32]: That's right, but it doesn't hurt to change the order and it makes more logical sense the other way.

File: geniza/corpus/tests/test_corpus_models.py (hunk 3 above)
Comment (blms) [reply geniza/corpus/tests/test_corpus_models.py]: Done in 476afbd 👍

File: geniza/corpus/tests/test_corpus_migrations.py [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+            change_message="created %s" % d,
+            action_flag=1,
+        )
+
+    def test_log_entries_reassigned(self):
+        # LogEntry should be reassigned so that its ContentType is corpus.Document
+        LogEntry = self.apps.get_model("admin", "LogEntry")
+        ContentType = self.apps.get_model("contenttypes", "ContentType")
+        document_type = ContentType.objects.get(app_label="corpus", model="document")
+        log_entry = LogEntry.objects.first()
```
[the remark this answers is not in the document]
Comment (blms) [reply geniza/corpus/tests/test_corpus_migrations.py]: Done in 6182d14 👍

File: geniza/corpus/migrations/0026_delete_documentprefetchableproxy.py (hunk 1 above)
Comment (blms) [reply geniza/corpus/migrations/0026_delete_documentprefetchableproxy.py]: Done in 92e4487 👍

**Pull request merged**